src/Common/src/Microsoft/Win32/SafeHandles/SafeEventStreamHandle.OSX.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Win32.SafeHandles
+{
+    /// <summary>
+    /// This class is a wrapper around the EventStream and Create pattern.
+    /// Usually, the Create pattern has the caller call Create* to allocate
+    /// and CFRelease to free; however, FSEventStream has it's own release
+    /// function, so we need to extend the pattern to account for that.
+    /// </summary>
+    [System.Security.SecurityCritical]
+    internal sealed partial class SafeEventStreamHandle : SafeHandle
+    {
+        internal SafeEventStreamHandle(IntPtr ptr) : base(IntPtr.Zero, true)
+        {
+            this.SetHandle(ptr);
+        }
+
+        [System.Security.SecurityCritical]
+        protected override bool ReleaseHandle()
+        {
+            Interop.EventStream.FSEventStreamStop(this);
```

Comment:
> I am uncomfortable sending the  SafeHandle  instance through the marshaller while in the middle of a call to  ReleaseHandle , but if others are sure this is fine then...

I'm actually surprised this worked (maybe it didn't?) I'd have expected it would fail during marshaling when attempting to addref.